src/Analyzers/Orchestration/OrchestrationAnalyzer.cs

```diff
@@ -316,7 +316,12 @@ protected virtual void VisitMethod(SemanticModel semanticModel, SyntaxNode metho
             IEnumerable<MethodDeclarationSyntax> calleeSyntaxes = calleeMethodSymbol.GetSyntaxNodes();
             foreach (MethodDeclarationSyntax calleeSyntax in calleeSyntaxes)
             {
-                this.FindInvokedMethods(semanticModel, calleeSyntax, calleeMethodSymbol, rootOrchestration, reportDiagnostic);
+                // Since the syntax tree of the callee method might be different from the caller method, we need to get the correct semantic model,
+                // avoiding the exception 'Syntax node is not within syntax tree'.
```

Comment:
Do we have a test that covers this scenario?

Reply:
Now I have haha